src/Cedar/Proto_PPP.c

```diff
@@ -807,14 +807,14 @@ bool PPPParseUsername(CEDAR *cedar, char *src_username, ETHERIP_ID *dst)
 	}
 	else
 	{
-		// Search for the last "@" in the string
+		// Search for the separator character's last position in the string

```

Comment:
in theory, there might be names like `chipitsine@gmail.com@mail.ru` ?
and they are not to be treated as error ?

Reply:
The `@` character is currently considered as invalid character in the username, I'm going to fix this behavior.

Reply:
I mean how to treat when multiple separators are found

Reply:
Oh, sorry.

In case there are multiple separators, only the last is used to split the string. In your example, the username would be `chipitsine@gmail.com` and the hub name `mail.ru`.

Reply:
can we some security issue here?

Reply:
what will happen if separator is the last character, i.e. `chipitsine@gmail.com@`

Reply:
We always copy after the separator character:
https://github.com/SoftEtherVPN/SoftEtherVPN/blob/b8629f324f81aff488f461468c4b44593b847513/src/Cedar/Proto_PPP.c#L837

However, it's not a problem because we check whether the hub exists:
https://github.com/SoftEtherVPN/SoftEtherVPN/blob/8a58af86b131b786a8b605851302cb1d541bfe78/src/Cedar/Proto_PPP.c#L839-L849